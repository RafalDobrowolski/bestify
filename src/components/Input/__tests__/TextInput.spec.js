import { mount } from '@vue/test-utils';
import bTextInput from "../TextInput";

describe('TextInput', () => {
  test('TextInput is rendered correctly', () => {
    const wrapper = mount(bTextInput);
    expect(wrapper.html()).toMatchSnapshot()
  });

  test('TextInput should emit an input event',  () => {
    const wrapper = mount(bTextInput);
    wrapper.find('input').trigger('input');
    expect(wrapper.emitted().input).toBeTruthy()
    expect(wrapper.emitted().input).toHaveLength(1);
  });

  test('TextInput is disabled', () => {
    const wrapper = mount(bTextInput, {
      propsData: {
        disabled: true
      }
    });
    expect(wrapper.find('input').attributes().disabled).toBeTruthy();
  });
});