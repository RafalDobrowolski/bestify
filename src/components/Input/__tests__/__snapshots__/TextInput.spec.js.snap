// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput TextInput is rendered correctly 1`] = `<input placeholder="" type="text" class="text-input">`;
